contribute/CONTRIBUTING-TO-CI-AND-E2E.md

```diff
@@ -1,112 +1,102 @@
+
+Here are some general contribution guidelines. 
+
+### Adding new tests/test-categories
+
+To add new tests, please raise an issue describing the objective, pre-requisites, test steps, expected results and cleanup routines.
+In case of proposals to add new test categories, add a generic description explaining the purpose and value addition
+
+Once accepted, the issue can be assigned to interested contributors who can make relevant changes with a pull request. 
+
+### Create new application YAMLs
+
+To add new applications to e2e, please consider the points made here, if applicable: 
+
+- https://kubernetes.io/docs/concepts/configuration/overview/
+- https://www.mirantis.com/blog/introduction-to-yaml-creating-a-kubernetes-deployment/
+- https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/
+
+These can be placed here : https://github.com/openebs/openebs/tree/master/k8s/demo
+
+### Create ansible playbooks to execute test workflows
+
+The test playbooks can be placed in e2e/ansible/playbooks/<test-category>.Here are some considerations. 
+
+- Each test playbook is recommended to be constructed with the following template.
+
+  The main test playbook, statically "includes" following files :
+
+  - Test variables YAML
+  - Pre-requisite tasks YAML
+  - Cleanup tasks YAML
+  
+  Auxiliary scripts that are invoked in the test playbook can be placed in the same directory. 
+
+ You can view a sample test playbook [here](https://github.com/openebs/openebs/tree/master/e2e/ansible/playbooks/hyperconverged/test-k8s-percona-mysql-pod)
+
+- Test playbooks that deploy applications are recommended to include steps to generate application-specific workloads that
+  will exercise the OpenEBS storage volume. 
+  
+- Here are some ansible best practices while constructing playbooks : 
+ 
+  - https://www.jeffgeerling.com/blog/yaml-best-practices-ansible-playbooks-tasks 
+  
+  The OpenEBS ansible environment setup procedure is provided [here](https://github.com/openebs/openebs/blob/master/e2e/ansible/openebs-on-premise-deployment-guide.md)
+  
+### Submit changes to fix bugs or improve e2e tests 
+
+Explain the fix in detail and add the verbose test playbook output that is run with the fixes (Run the playbook with a -v option for this). Also, consider the general ansible playbook best practices referenced in the previous section.
+
+### Raise or fix issues on the CI framework 
+
+As with other issues, please mention the bug/enhancements in detail with logs/references if possible.
+
+The jenkins configuration used to setup OpenEBS e2e can be found [here](https://github.com/openebs/openebs/blob/master/e2e/jenkins/README.md)
+
+### Raise or fix issues on the e2e, CI documentation
+
+Getting Documentation Right is Hard! Please raise a PR with you proposed changes to the READMEs, tutorials and deployment guides in openebs/e2e
+
+If you need any help with git, refer to this [cheat sheet](./contribute/git-cheatsheet.md).
```

Comment:
Git

Reply:
I would prefer for it to be smallcase. Since the official site uses it as the same if it is not violating the grammatical sense.